Change sink worker to allow for multiple queues in config

martinsumner opened the pull request:
Peer configuration can now contain a queue-name, and hence we can have mutliple queues in the configuration.

Also allows for the worker count to be changed for a given queue.

There is now an iteration number kept - to make it easier when switching configurations to discard in-flight work items that relate to old configuration.

This change means that protocol no longer defaults to 'http' - it must be explicity set.

ThomasArts committed:
Adapt to new peers format

martinsumner committed:
Change sink worker to allow for multiple queues in config

Peer configuration can now contain a queue-name, and hence we can have mutliple queues in the configuration.

Also allows for the worker count to be changed for a given queue.

There is now an iteration number kept - to make it easier when switching configurations to discard in-flight work items that relate to old configuration.

This change means that protocol no longer defaults to 'http' - it must be explicity set.

ThomasArts commented on src/riak_kv_replrtq_snk.erl:
```diff
@@ -339,25 +381,39 @@ calc_mean(Time, Count) ->
 % usable peer information.
 % tokenised string expected to be of form:
 % "192.168.10.1:8098|192.168.10.2:8098 etc"
```
should we not have a name here in the comment as well?

martinsumner replied:
Updated comment

ThomasArts committed:
Adapt to queue part of config format

ThomasArts committed:
Reduce number workers to max 5

If we have more than 5 workers, it seems we get on average too few workers taking work, but tests are not long enough to get accuracy in that measurement

martinsumner committed:
Merge pull request #1740 from Quviq/mas-i1691-snkworker

QuickCheck model for snkworker changes

martinsumner committed:
Add explanation to set_workercount/2

As per feedback from @thomasarts - nee to explain semantics of this.

martinsumner committed:
Update comment on tokenised string
